src/circuit/auto-relay.js

```diff
@@ -21,14 +21,25 @@ const {
   RELAY_RENDEZVOUS_NS
 } = require('./constants')
 
+/**
+ * @typedef {import('libp2p-interfaces/src/connection').Connection} Connection
+ * @typedef {import('../peer-store/address-book').Address} Address
+ */
+
+/**
+ * @typedef {Object} AutoRelayProperties
```

Comment:
Nit: In js-ipfs we settled on calling injected deps with `Config` suffix e.g. AutoRelayConfig`. Might be worth following same convention across both code bases.

Reply:
Probably mentioning in the one below, right?
The current approach is: `xProperties` to libp2p references needed and `xOptions` for user options, so that we can use `xOptions` on the top level config docs. We can definitely change `xOptions` to `xConfig` and I would be happy in having a better name for `xProperties` if you have any suggestions

Reply:
Bo I was referring to `xProperties` actually, maybe I’ve misread the code, but it looked like dependencies injected into `AutoRelay` and we called such things with `xConfig` e.g.:

https://github.com/ipfs/js-ipfs/blob/530767d85d85b5932abdb87721e395c0a214b148/packages/ipfs-core/src/components/dag/get.js#L8-13

Reply:
I would like to get a good way for here. So, in the libp2p stand point we have lots of places where we inject libp2p, or some libp2p components on other libp2p components. Moreover, these libp2p components have configurations that users can modify.
We need to differentiate between them. My initial approach was to make `ComponentNameProperties` for the libp2p components we provide, and `ComponentNameOptions` for user options. Naming config for dependencies that are not "configurable" does not seem the best naming to me. However, I am not totally happy with `xProperties`, so if we can find a better convention would be great

Reply:
Maybe `AutoRelayComponents` then ? 

The reason I end up calling those config in js-ipfs is because it is configuration of the component. It is true that user can't modify those but who ever creates a component can provide alternative configuration e.g. in memory repo vs on disk repo etc...

Things that user can configure are referred as `PrefixOptions`.


I don't have strong feeling about naming convention itself, just would love to settle on the same convention across ipfs and libp2p

Reply:
Components seems a good alternative. Will wait for other people to weight in before renaming everywhere

Reply:
yeah I agree. But let's keep the discussion open until hugo/jacob comments